charts/hivemq-platform/templates/_helpers.tpl

```diff
@@ -475,6 +474,82 @@ Usage: {{- include "hivemq-platform.validate-additional-volumes" . }}
 {{- end -}}
 {{- end -}}
 
+{{/*
+Validates the broker license, additional broker licenses, the enterprise extension licenses and the Data Hub license so:
+ - When the `license.create` value is enabled, a least one license content is defined.
+ - Only one of `data` or `overrideLicense` are set, and not both.
+ - No duplicated "license.lic" Secret entries is found for additional broker licenses.
+Usage: {{ include "hivemq-platform.validate-licenses" . }}
+*/}}
+{{- define "hivemq-platform.validate-licenses" -}}
+{{- $licenseContentExists := "" -}}
+{{- if .Values.license.create -}}
+    {{- include "hivemq-platform.validate-license-content" (dict "licenses" .Values.license.additionalLicenses "licenseLabel" "Additional HiveMQ Broker") -}}
+    {{- include "hivemq-platform.validate-license-content" (dict "licenses" .Values.license.extensions "licenseLabel" "HiveMQ Enterprise Extension") -}}
+    {{- include "hivemq-platform.validate-license-content" (dict "licenses" .Values.license.dataHub "licenseLabel" "HiveMQ Data Hub") -}}
+    {{- if and (not .Values.license.overrideLicense) (not .Values.license.data) (not .Values.license.additionalLicenses) (not .Values.license.extensions) (not .Values.license.dataHub) -}}
+        {{- fail (printf "HiveMQ Platform license content cannot be empty") -}}
+    {{- end -}}
+    {{- if and .Values.license.data .Values.license.overrideLicense -}}
+        {{- fail (printf "Both `data` and `overrideLicense` values are set for the HiveMQ Broker license content. Please, use only one of them") -}}
+    {{- end -}}
+    {{- include "hivemq-platform.validate-duplicated-additional-broker-license" . -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Validates the licenses content within a range function:
+ - At least either `data` or `overrideLicense` is not empty.
+ - Only one of `data` or `overrideLicense` are set, and not both.
+Usage: {{ include "hivemq-platform.validate-license-content" (dict "licenses" .licenses "licenseLabel" "HiveMQ Enterprise Extension") }}
+*/}}
+{{- define "hivemq-platform.validate-license-content" -}}
+{{- $label := .licenseLabel -}}
+{{- range $licenseName, $license := .licenses -}}
+    {{- if and (not $license.data) (not $license.overrideLicense) -}}
```

Comment:
I think the checks in this method are overlapping a bit. For example, when I configure just the key `foobar` it will print the `cannot be empty` error message, although the `Invalid values` would be more fitting.